executor/runtime/docker/docker.go

```diff
@@ -1501,37 +1543,57 @@ func getMainContainerRoot(inspectOutput types.ContainerJSON) string {
 	return inspectOutput.GraphDriver.Data["MergedDir"]
 }
 
-func (r *DockerRuntime) statusMonitor(cancel context.CancelFunc, c runtimeTypes.Container, eventChan <-chan events.Message, errChan <-chan error, statusMessageChan chan runtimeTypes.StatusMessage) {
-	defer close(statusMessageChan)
-	defer cancel()
+func (r *DockerRuntime) getExtraContainerIDs() []string {
+	ids := []string{}
+	for _, c := range r.c.ExtraPlatformContainers() {
+		ids = append(ids, c.Status.ContainerID)
+	}
+	for _, c := range r.c.ExtraUserContainers() {
+		ids = append(ids, c.Status.ContainerID)
+	}
+	return ids
+}
 
-	// This context should be tied to the lifetime of the container -- it will get significantly less broken
-	// when we tear out the launchguard code
+func isClosed(ch <-chan runtimeTypes.StatusMessage) bool {
```

Comment:
This is wrong.
I'm not using this channel correctly (reading from it when testing if it is closed).

I'll redo.